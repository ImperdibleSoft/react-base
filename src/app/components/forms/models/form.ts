import { FormState } from '../types/form';

export const getState = (): FormState => ({
  canBeLocked: true,
  hasPermissions: false,
  isEditEnabled: false,
});
