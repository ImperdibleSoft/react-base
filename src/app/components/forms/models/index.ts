import { FieldTypes, PersistedSmartForm, SmartFormState } from '../types';

export const TEXT: FieldTypes = 'text';
export const EMAIL: FieldTypes = 'email';
export const NUMBER: FieldTypes = 'number';
export const PASSWORD: FieldTypes = 'password';
export const TEXTAREA: FieldTypes = 'textarea';
export const SWITCH: FieldTypes = 'switch';
export const CHECKBOX: FieldTypes = 'checkbox';
export const DROPDOWN: FieldTypes = 'dropdown';
export const DATE: FieldTypes = 'date';
export const BUTTON: FieldTypes = 'button';
export const SUBMIT: FieldTypes = 'submit';
export const ARRAY: FieldTypes = 'array';

const initForm = (): PersistedSmartForm => {
  return {
    state: 'pristine',
    errors: [],
  };
};

export const getState = (): SmartFormState => ({
  form: initForm(),
});
