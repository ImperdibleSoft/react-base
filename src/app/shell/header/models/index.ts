import { HeaderMenu } from '../types';

export const noMenu: HeaderMenu = '';
export const modulesMenu: HeaderMenu = 'modules';
export const companiesMenu: HeaderMenu = 'companies';
export const userMenu: HeaderMenu = 'user';
