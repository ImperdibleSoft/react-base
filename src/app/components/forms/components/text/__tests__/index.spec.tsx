import * as React from 'react';
import { shallow } from 'enzyme';

import Text from '..';
import { EMAIL, NUMBER, PASSWORD, TEXT, TEXTAREA } from '../../../models';

describe('Text', () => {
  it('should render a valid email field', () => {
    const options = {
      id: 'email-field',
      label: 'Field label',
      type: EMAIL,
    };

    const component = shallow(<Text onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid number field', () => {
    const options = {
      id: 'number-field',
      label: 'Field label',
      type: NUMBER,
    };

    const component = shallow(<Text onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid password field', () => {
    const options = {
      id: 'password-field',
      label: 'Field label',
      type: PASSWORD,
    };

    const component = shallow(<Text onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid text field', () => {
    const options = {
      id: 'text-field',
      label: 'Field label',
      type: TEXT,
    };

    const component = shallow(<Text onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid textarea field', () => {
    const options = {
      id: 'text-field',
      label: 'Field label',
      type: TEXTAREA,
    };

    const component = shallow(<Text onChange={() => {}} onClick={() => {}} onFocus={() => {}} options={options} />);

    expect(component).toMatchSnapshot();
  });
});
