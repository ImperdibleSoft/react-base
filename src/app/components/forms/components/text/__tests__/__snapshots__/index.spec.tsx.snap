// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render a valid email field 1`] = `
<Wrapper>
  <Description>
    <Label
      htmlFor="email-field"
    >
      <Content
        options={
          Object {
            "id": "email-field",
            "label": "Field label",
            "type": "email",
          }
        }
      />
    </Label>
  </Description>
  <Input
    defaultValue=""
    id="email-field"
    name="email-field"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="email"
  />
</Wrapper>
`;

exports[`Text should render a valid number field 1`] = `
<Wrapper>
  <Description>
    <Label
      htmlFor="number-field"
    >
      <Content
        options={
          Object {
            "id": "number-field",
            "label": "Field label",
            "type": "number",
          }
        }
      />
    </Label>
  </Description>
  <Input
    defaultValue=""
    id="number-field"
    name="number-field"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="number"
  />
</Wrapper>
`;

exports[`Text should render a valid password field 1`] = `
<Wrapper>
  <Description>
    <Label
      htmlFor="password-field"
    >
      <Content
        options={
          Object {
            "id": "password-field",
            "label": "Field label",
            "type": "password",
          }
        }
      />
    </Label>
  </Description>
  <Input
    defaultValue=""
    id="password-field"
    name="password-field"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="password"
  />
</Wrapper>
`;

exports[`Text should render a valid text field 1`] = `
<Wrapper>
  <Description>
    <Label
      htmlFor="text-field"
    >
      <Content
        options={
          Object {
            "id": "text-field",
            "label": "Field label",
            "type": "text",
          }
        }
      />
    </Label>
  </Description>
  <Input
    defaultValue=""
    id="text-field"
    name="text-field"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="text"
  />
</Wrapper>
`;

exports[`Text should render a valid textarea field 1`] = `
<Wrapper>
  <Description>
    <Label
      htmlFor="text-field"
    >
      <Content
        options={
          Object {
            "id": "text-field",
            "label": "Field label",
            "type": "textarea",
          }
        }
      />
    </Label>
  </Description>
  <TextareaAutosize
    defaultValue=""
    id="text-field"
    inputRef={[Function]}
    name="text-field"
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHeightChange={[Function]}
    style={
      Object {
        "appearance": "none",
        "backgroundColor": "#fff",
        "border": "none",
        "borderBottom": "1px solid #779ec1",
        "borderRadius": "4px",
        "borderTop": "1px solid transparent",
        "color": "inherit",
        "cursor": "inherit",
        "display": "block",
        "fontFamily": "inherit",
        "fontSize": "inherit",
        "lineHeight": "1.3em",
        "minHeight": "150px",
        "padding": "5px 10px",
        "resize": "vertical",
        "transition": "background-color 0.2s, border: 0.2s, color 0.2s",
        "userSelect": "initial",
        "width": "100%",
      }
    }
    useCacheForDOMMeasurements={false}
  />
</Wrapper>
`;
